File: get_appdir.py
import os
from appdirs import user_data_dir
from dotenv import load_dotenv

load_dotenv()

APPNAME = os.getenv("APPNAME")
APPAUTHOR = os.getenv("APPAUTHOR")


def get_appdir():
    return user_data_dir(APPNAME, APPAUTHOR)

File: get_appdir.ts
import 'dotenv/config'
import { panic } from 'functional-utilities'

export async function get_appdir(): Promise<string> {
    const { python } = await import('pythonia')
    const get_appdir_py = await python('appdirs')
    const appdir = await get_appdir_py.user_data_dir(
        process.env.APPNAME ?? panic('APPNAME is not set'),
        process.env.APPAUTHOR ?? panic('APPAUTHOR is not set')
    )
    ;(python as any).exit()
    return appdir
}

console.log(await get_appdir())
